### PR TITLE
fix itemAdapter 1.18.2

### PR DESCRIPTION
**Pull request by J3fftw1**
Fix for https://pastebin.com/100WXjwW

Apparently the nms changed again in 1.18.2.

This would make it work on 1.18.2 but it would fail at 1.18 and 1.18.1 do we want to separately check that or is this sufficiënt 

My commit name is a little of i thought at first it was for .1.18.1 aswell

### CODE REVIEW

**Comment by WalshyDev** [comment]
Yeah don't just break 1.18 and 1.18.1. Make a new class for 1.18.2 specifically

**Comment by J3fftw1** [comment]
How do you guys feel about the class name?

**Comment by md5sha256** [comment]
I think this is probably fine. However, do we have a policy as to what versions we support/when we drop support for legacy minor versions?

**Comment by WalshyDev** [comment]
> However, do we have a policy as to what versions we support/when we drop support for legacy minor versions?

We do not right now. It isn't been bad to support 1.14-1.18 so far. Though, if we see NMS changes on patch versions then it may be something to consider more.

---

Regardless, this PR looks good. Do we need any changes to the other NMS classes or is this the only one affected?

**Comment by J3fftw1** [comment]
from the report i think its just this

**Comment by WalshyDev** [comment]
Please update the PR description. Tests should also be done to validate the other cases are ok.

File: dough-items/src/main/java/io/github/bakedlibs/dough/items/nms/ItemNameAdapter18_2.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import io.github.bakedlibs.dough.reflection.ReflectionUtils;
+import io.github.bakedlibs.dough.versions.UnknownServerVersionException;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class ItemNameAdapter18_2 implements ItemNameAdapter {
```
Comment (TheBusyBiscuit) [inline dough-items/src/main/java/io/github/bakedlibs/dough/items/nms/ItemNameAdapter18_2.java:11]:
```suggestion
class ItemNameAdapter18v2 implements ItemNameAdapter {
```

File: dough-items/src/main/java/io/github/bakedlibs/dough/items/nms/ItemNameAdapter18_2.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    private final Method getCopy;
+    private final Method getName;
+    private final Method toString;
+
+    ItemNameAdapter18_2() throws NoSuchMethodException, SecurityException, ClassNotFoundException, UnknownServerVersionException {
```
Comment (TheBusyBiscuit) [inline dough-items/src/main/java/io/github/bakedlibs/dough/items/nms/ItemNameAdapter18_2.java:17]:
```suggestion
    ItemNameAdapter18v2() throws NoSuchMethodException, SecurityException, ClassNotFoundException, UnknownServerVersionException {
```

File: dough-items/src/main/java/io/github/bakedlibs/dough/items/nms/ItemNameAdapter.java [hunk 3]
```diff
@@ -25,6 +25,8 @@
             if (MinecraftVersion.isMocked()) {
                 // Special case for MockBukkit
                 return new ItemNameAdapterMockBukkit();
+            } else if (version.isAtLeast(1, 18, 2)) {
+                return new ItemNameAdapter18_2();
```
Comment (TheBusyBiscuit) [inline dough-items/src/main/java/io/github/bakedlibs/dough/items/nms/ItemNameAdapter.java:29]:
```suggestion
                return new ItemNameAdapter18v2();
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
Using an underscore in a class name is not exactly recommended iirc.
We could just use `v` instead here.
Anyway, also spotted a minor mistake with the class visibility, otherwise it looks good to me.
Thanks!

**Review by TheBusyBiscuit — APPROVED** [review]
LGTM
@WalshyDev You do a review too and feel free to merge when ready. I will publish a new dough version asap then

**Review by md5sha256 — APPROVED** [review]
LGTM

**Comment by md5sha256** [comment]
Although I just remembered that something also broke for the adapter in `dough-skins` from 1.17 to 1.18. Not sure if its affected by this minor revision change but worth double-checking I guess

**Comment by md5sha256** [comment]
Just did a quick check, looks like the net.minecraft method is still the same for the adapter in dough-skins so this PR should have everything for 1.18. However, we do need to address the migration of the IS repo (mvn.intellectualsites.com)

**Comment by WalshyDev** [comment]
Merging, then we'll get @md5sha256's in and release

**Pull request merged**